services/meta/service.go

```diff
@@ -108,21 +108,25 @@ func (s *Service) serve() {
 // Close closes the underlying listener.
 func (s *Service) Close() error {
 	if s.ln != nil {
-		return s.ln.Close()
-	}
-
-	if err := s.store.close(); err != nil {
-		return err
+		if err := s.ln.Close(); err != nil {
+			return err
+		}
 	}
+	s.handler.Close()
```

Comment:
I think so. Because the open connections in the handler are using the store. So we need it to close first so we know that it's safe.